### PR TITLE
TTD: Add support for innerloop Time-Travel Debugging.

### PR DESCRIPTION
**Pull request by mrkmarron**
This PR adds flags to enable trace recording with the debugger attached and allows for a debugger to write a trace to disk. 

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

TTD

### CODE REVIEW

**Review by mike-kaufman — APPROVED** [review]
lgtm.  Looks like there's no deps on an updated chakra-core here, but if you can confirm that'd be good.

**Comment by mrkmarron** [comment]
Hi Mike, there is one place that has a dependency on a new JSRTDebug API. So, we will need to stage with that.

File: deps/chakrashim/src/jsrtisolateshim.cc [hunk 1]
```diff
@@ -649,9 +649,7 @@ bool IsolateShim::RunSingleStepOfReverseMoveLoop(v8::Isolate* isolate,
                                                  int64_t* nextEventTime) {
     int64_t snapEventTime = -1;
     int64_t snapEventEndTime = -1;
-    int64_t origNETime = *nextEventTime;
-    // TODO(mrkmarron) JsTTDMoveMode is no longer 64 bit wide, fix use of
```
Comment (sethbrenith) [inline deps/chakrashim/src/jsrtisolateshim.cc:653]: Is this comment still relevant? I don't see any other changes to JsTTDMoveMode in this review.

**Review by sethbrenith — APPROVED** [review]

File: deps/chakrashim/src/jsrtisolateshim.cc (hunk 1 above)
Comment (mrkmarron) [reply deps/chakrashim/src/jsrtisolateshim.cc:653]: This comment was stale so I wanted to clean it up.

File: deps/chakrashim/src/inspector/js_protocol.json [hunk 2]
```diff
@@ -782,6 +782,13 @@
         "domain": "TimeTravel",
         "description": "TimeTravel domain exposes JavaScript time travel capabilities. It allows stepping backwards through execution.",
         "commands": [
+            {
+              "name": "writeTTDLog",
```
Comment (kfarnung) [inline deps/chakrashim/src/inspector/js_protocol.json:786]: nit:  The spacing is a little off here

**Review by kfarnung — APPROVED** [review]

**Comment by jackhorton** [comment]
Ran into this in today's merge pump. I will try to cherry-pick this commit and work from there.

**Comment by jackhorton** [comment]
Landed in dc759ca325c888f203fb3d4f15cebd74d2f1427b

**Pull request closed without merging**